subprojects/core/src/main/java/org/gradle/api/internal/attributes/AttributeValue.java

```diff
@@ -59,14 +56,20 @@ public Object get() {
 
     /**
      * Returns the value of this attribute.
+     * <p>
+     * This should <strong>NOT</strong> be called on {@link AttributeValue#MISSING}.
```

Comment:
```suggestion
     * @throws UnsupportedOperationException if this is called on a non-{@link #isPresent present} value.
```

Javadoc should describe the interface, not implementations of the interface. 

Applies below as well.